### DISCUSSION TITLE
TID error

### DISCUSSION THREAD

**Discussion by gbusoni** [FeynCalc/feyncalc#162] [General]
I was using Feyncalc some time ago. I struggle using the new syntax. I am unable to make the following code to work

```
Den = FeynAmpDenominator[PropagatorDenominator[Momentum[k, 4], M], 
  PropagatorDenominator[Momentum[k - p, 4], m], 
  PropagatorDenominator[Momentum[k + q - p, 4], m]]
Num = SpinorUBar[p - q, 
   0].(GS[p - k - q] - m).GA[\[Mu]].(GS[p - k] - m).SpinorU[p, m]
Int = Num*Den /. q -> p - l // FCI (*Does not seem to like SpinorU[p-q]*)
OneLoopSimplify[Int, k]
```

OutPut:

> TID::failmsg: Error! TID has encountered a fatal problem and must abort the computation. The problem reads: Your input contains a mixture of 4- and D-dimensional quantities. This is in general not allowed in dimensional regularization, unless you are using the Breitenlohner-Maison-t'Hooft-Veltman scheme.

**Comment by vsht** [comment]
It is what the error message tells you: You are trying to do a D-dimensional calculation
with 4-dimensional quantities. 

Older versions of FeynCalc used to silently "upgrade" 4-dim objects to D-dim ones, 
but this only lead to more confusion and unexpected behavior. So the current philosophy
is to take the user input as it is and abort the calculation if the input doesn't make sense.

Changing everything to D-dimensions in your example works fine

```
Den = FAD[{k, M}, {k - p, m}, {k + q - p, m}]
Num = SpinorUBarD[p - q, 0] . (GSD[p - k - q] - m) . 
  GAD[\[Mu]] . (GSD[p - k] - m) . SpinorUD[p, m]
Int = Num*Den /. q -> p - l // 
  FCI (*Does not seem to like SpinorU[p-q]*)
OneLoopSimplify[Int, k]
```
